sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -130,6 +130,13 @@ class StreamExecution(
   protected var offsetSeqMetadata = OffsetSeqMetadata(
     batchWatermarkMs = 0, batchTimestampMs = 0, sparkSession.conf)
 
+  /**
+   * A map of current watermarks, keyed by the position of the watermark operator in the
+   * physical plan. The minimum watermark timestamp present here will be used and persisted as the
+   * query's watermark when preparing each batch, so it's ok that this val isn't fault-tolerant.
```

Comment:
Make it clear that "this state is soft state and does not require persistence, because the minimum .....